### ISSUE TITLE
Generated strings not rendering correctly in new resource loading mechanism

### ISSUE THREAD

**Issue by mshearer123** [cashapp/paparazzi#1067]
**Description**
After updating to paparazzi 1.3.1, strings do not render from R.strings correctly anymore.
using `stringResource(id = R.string.x)`

We use compose public Previews in our paparazzi tests.   The strings render fine in the preview.

**Steps to Reproduce**
Provide a sample project, failing test, or steps to reproduce.

**Expected behavior**
String would be rendered from R resources correctly

**Additional information:**
- Paparazzi Version: 1.3.1
- OS: mac (m1)
- Compile SDK: 33
- Gradle Version: 8.0.2
- Android Gradle Plugin Version: 8.0.2

**Screenshots**
![Screenshot 2023-08-30 at 10 23 17](https://github.com/cashapp/paparazzi/assets/2699109/a44ced6a-f28f-467b-8595-8abc4f2758fa)

**update**
fixed using
```
app.cash.paparazzi.legacy.resource.loading=true
app.cash.paparazzi.legacy.asset.loading=true
```

**Comment by jrodbx** [comment]
Closed by #1307

**Comment by mshearer123** [comment]
This issue has resurfaced for us in v1.3.3
we had been using
```
app.cash.paparazzi.legacy.resource.loading=false
app.cash.paparazzi.legacy.asset.loading=false
```

**Comment by kevinzheng-ap** [comment]
@mshearer123 I can definitely help to look into more details. 

1. Could you provide how these generated strings are declared in your gradle file?
2. Could you check if the generated string are in `build/generated/res/resValues/debug/values/gradleResValues.xml`? And if `build/generated/res/resValues/debug/values` exists in `build/intermediates/paparazzi/debug/resources.json`?

**Comment by mshearer123** [comment]
Thanks @kevinzheng-ap ,

We don't have `gradleResValues.xml` files, we have a translation task that generates

**Comment by kevinzheng-ap** [comment]
@mshearer123 How do you ask gradle to use the resources from these files in your task?

**Comment by mshearer123** [comment]
i believe with these functions
```
  @TaskAction
  fun generateStringsXml() {
    val englishMap = Yaml().load<Map<String, Any>>(File("${langFolder.get().asFile.absolutePath}/en.lyaml").inputStream())
    val reference = englishMap["en"] as LinkedHashMap<String, Any>

    listOf(
      "en" to "values",
      "da" to "values-da",
      "de" to "values-de",
      "es" to "values-es",
      "fi" to "values-fi",
      "fr" to "values-fr",
      "it" to "values-it",
      "ja" to "values-ja",
      "nl" to "values-nl",
      "no" to "values-nb",
      "no" to "values-nn", //We're including no twice to support the nb and nn variations of Norwegian.
      "pl" to "values-pl",
      "pt-br" to "values-pt",
      "sv" to "values-sv",
      "zh-tw" to "values-zh",
      "zh-tw" to "values-zh-rTW", //We're including zh-tw twice to use it for Traditional and Simplified Chinese.
    ).forEach { (input, resourcesFolder) -> parse("$input.lyaml", "${outputFolder.get().asFile.absolutePath}/$resourcesFolder", reference) }
  }
```

```
  private fun parse(filename: String, outputFolder: String, reference: LinkedHashMap<String, Any>) {
    val lyamlFile = File("${langFolder.get().asFile.absolutePath}/$filename")
    if (!lyamlFile.exists()) return

    val map = Yaml().load<Map<String, Any>>(lyamlFile.inputStream())
    val lang = map.keys.firstOrNull() ?: return

    val strings = parse(node = map[lang] as LinkedHashMap<String, Any>, referenceNode = reference)
    File(outputFolder).mkdirs()
    File(outputFolder, "generated_strings.xml").printWriter().use { file ->
      file.println("<?xml version=\"1.0\" encoding=\"utf-8\"?>")
      file.println("<resources>")
      strings.forEach(file::println)
      file.println("</resources>")
    }
  }
```

**Comment by kevinzheng-ap** [comment]
These functions look like how resources XML files are generated. We need more info about how they are used as custom resources in gradle. Do you change `sourceSets` or something else in gradle file to include the `outputFolder` ?

**Comment by mshearer123** [comment]
ah sorry,

when registering the task 
```
    val resOutputDir = File(project.buildDir.path, "/generated/res/i18n/main")
    val i18nXmlTask = tasks.register<I18nXmlGeneratorTask>("i18nXml") {
      group = "hubspot"
      langFolder.set(inputFolder)
      outputFolder.set(resOutputDir)
    }
```

**Comment by mshearer123** [comment]
whole class 

```
class LocalizationConventionPlugin : ConventionPlugin {
  override fun Project.configure() {
    val inputFolder = File("${project.projectDir}/lang")
    if (!inputFolder.exists()) return

    val bethBotTask = tasks.register<BethBotTask>("bethbot") {
      langFolder.set(inputFolder)
      outputFolder.set(File(project.buildDir.path, "/bethbot"))
    }

    val i18nKotlinTask = tasks.register<I18nKotlinGeneratorTask>("i18nKotlin") {
      group = "hubspot"
      langFolder.set(inputFolder)
      flags.set("kotlin")
    }

    val resOutputDir = File(project.buildDir.path, "/generated/res/i18n/main")
    val i18nXmlTask = tasks.register<I18nXmlGeneratorTask>("i18nXml") {
      group = "hubspot"
      langFolder.set(inputFolder)
      outputFolder.set(resOutputDir)
    }

    i18nKotlinTask.dependsOn(bethBotTask)
    i18nXmlTask.dependsOn(bethBotTask)

    // consider splitting this in 3 different conventions (base, androidApp, android Lib)
    when (val androidExtension = project.extensions.getByName("android")) {
      is com.android.build.gradle.AppExtension -> androidExtension.applicationVariants.all {
        configureTasks(this@configure, this, i18nKotlinTask, resOutputDir, i18nXmlTask)
      }

      is com.android.build.gradle.LibraryExtension -> androidExtension.libraryVariants.all {
        configureTasks(this@configure, this, i18nKotlinTask, resOutputDir, i18nXmlTask)
      }
    }
  }

  private fun configureTasks(
    project: Project,
    baseVariant: BaseVariant,
    i18nKotlinTask: TaskProvider<I18nKotlinGeneratorTask>,
    resOutputDir: File,
    i18nXmlTask: TaskProvider<I18nXmlGeneratorTask>,
  ) {

    val pkgName = baseVariant.applicationId.removeSuffix(baseVariant.buildType.applicationIdSuffix.orEmpty())
    val outputSrc = File(project.project.buildDir.path, "/generated/source/i18n/main/${pkgName}")
    i18nKotlinTask.configure {
      outputFolder.set(outputSrc)
      packageName.set(pkgName)
    }
    baseVariant.registerJavaGeneratingTask(i18nKotlinTask, outputSrc)
    baseVariant.registerGeneratedResFolders(project.project.files(resOutputDir).builtBy(i18nXmlTask))
    baseVariant.mergeAssetsProvider.dependsOn(i18nXmlTask)
  }
}
```

**baseVariant.registerGeneratedResFolders(project.project.files(resOutputDir).builtBy(i18nXmlTask))**

**Comment by kevinzheng-ap** [comment]
`baseVariant.registerGeneratedResFolders(project.project.files(resOutputDir).builtBy(i18nXmlTask))` is what I am looking for. Will raise a PR when additional fixes are ready

**Comment by mshearer123** [comment]
amazing, thanks @kevinzheng-ap

**Comment by kevinzheng-ap** [comment]
@mshearer123 PR has been merged and please try it again on 1.3.4
